Description: Select appropriate linker script for kFreeBSD-amd64
 Enable building rEFIt on kfreebsd-amd64.
Author: Julien BLACHE <jblache@debian.org>

Index: refit-0.14/Make.common
===================================================================
--- refit-0.14.orig/Make.common	2010-04-19 18:30:56.377932378 +0200
+++ refit-0.14/Make.common	2010-04-19 18:30:52.542932839 +0200
@@ -9,6 +9,7 @@
 EFICRT0         = /usr/lib
 
 ARCH            = $(shell uname -m | sed s,i[3456789]86,ia32,)
+OS		= $(shell uname -s)
 CPPFLAGS        = -I$(EFIINC) -I$(EFIINC)/$(ARCH) -I$(EFIINC)/protocol -DCONFIG_$(ARCH)
 
 OPTIMFLAGS      = -O2 -fno-strict-aliasing
@@ -31,7 +32,16 @@
 
 
 CRTOBJS         = $(EFICRT0)/crt0-efi-$(ARCH).o
-LDSCRIPT        = $(EFICRT0)/elf_$(ARCH)_efi.lds
+
+ifneq (,$(findstring FreeBSD,$(OS)))
+ ifeq ($(ARCH),x86_64)
+	LDSCRIPT = $(EFICRT0)/elf_$(ARCH)_fbsd_efi.lds
+ else
+	LDSCRIPT = $(EFICRT0)/elf_$(ARCH)_efi.lds
+ endif
+else
+	LDSCRIPT = $(EFICRT0)/elf_$(ARCH)_efi.lds
+endif
 
 LDFLAGS        += -T $(LDSCRIPT) -shared -Bsymbolic -L$(EFILIB) -L$(GNUEFILIB) $(CRTOBJS)
 LIBS            = -lefi -lgnuefi $(shell $(CC) -print-libgcc-file-name)
